libnavigation-core/build.gradle

```diff
@@ -54,6 +54,9 @@ dependencies {
     implementation dependenciesList.coroutinesAndroid
     implementation dependenciesList.androidStartup
 
+    //TODO: move to a separate module before merge
+    implementation dependenciesList.mapboxSdkServices
```

Comment:
I'm not sure if additional dependency in core module is so bad so that we need a new module. We have this dependency in the router module, and as I know we haven't ever heart from customers that they are using custom router. @dzinad , @DzmitryFomchyn , what is your opinion on that?

Reply:
I don't think that this will affect a lot of customers, but:
1) I'm not sure that adding a dependency is OK for a patch (libnavigation-core is a separate dependency after all);
2) More importantly, it breaks the idea of a custom router. This API can be used by anyone. The fact that we haven't heard of anyone using a custom router doesn't mean nobody does that. We thought nobody needed reroute controller, but once we deprecated it we started receiving complaints. And here I'm not only worried about adding a new unnecessary dependency for them, but rather breaking their logic: if they unexpectedly start receiving routes from an unknown (for them) router, it might have unpredictable effect. I think anyone would expect the routes to come from the provided router. And it makes sense. I think it might also affect billing (I'm not 100% sure how that works, won't they get charged if their token starts doing route requests to our server that they're unaware of?)

Of course, it the API is used only by one customer that we know son't have any problems with that, it's fine. But it's not stable: this customer changes something - it might be broken, someone else starts using it (and it's not prohibited in any way) - it might be broken. Then it will become very difficult to support. And I'm worried about that.
I would vote for using the router. If we want to adjust our router's logic in this case: OK, let's provide another router (that does practically the same thing you implemented here). But if the router is custom, let's use it.

To be more specific, I suggest:
1. along with `moduleRouter` add a new `offlineToOnlineSwitchRouter` field;
2. set it the same as `moduleRouter` for a custom router;
3. set it to the new implementation added in this PR if it's our router.

I know it seems like an overhead. But you never know. We added a lot of customization opportunities earlier and now we have to live with it, even if we think now they're unnecessary or unused.

Reply:
agreed to implement custom always online router later